src/main/scala/com/linkedin/spark/datasources/tfrecord/DefaultSource.scala

```diff
@@ -33,13 +33,25 @@ class DefaultSource extends FileFormat with DataSourceRegister {
       options: Map[String, String],
       files: Seq[FileStatus]): Option[StructType] = {
     val recordType = options.getOrElse("recordType", "Example")
-    // Use the first non-empty file
-    // For smaller dataset, it is possible the first a few part files have zero length
-    val nonZeroIndex = files.indexWhere(f => f.getLen > 0)
-    val fileIndex = if (nonZeroIndex > 0) nonZeroIndex else 0
-    val rdd = sparkSession.sparkContext.newAPIHadoopFile(files(fileIndex).getPath.toString,
+    files.collectFirst {
+      case f if hasSchema(sparkSession, f, recordType) => getSchemaFromFile(sparkSession, f, recordType)
+    }
```

Comment:
Is there any reason to use `collectFirst`? Can we just use any non-empty file to get the schema?

Reply:
The original code was checking file size, but that's not enough. That's why we get issue #27 .
For compressed files, due to the header, a file with empty content has non-zero file size.
This RB enhanced it by checking beyond the file size.

Reply:
I see, thanks for the explanation.